### PR TITLE
Properly set config to None, and error on environment setup

### PR DESCRIPTION
**Pull request by ericholscher**

### CODE REVIEW

File: readthedocs/projects/tasks.py [hunk 1]
```diff
@@ -127,6 +128,10 @@ def run(self, pk, version_pk=None, build_pk=None, record=True, docker=False,
 
             self.config = load_yaml_config(version=self.version)
 
+        if self.setup_env.failed:
+            self.send_notifications()
+            raise BuildEnvironmentError(_('Build Setup Failed'))
```
Comment (ericholscher) [inline readthedocs/projects/tasks.py]: Not sure if it's better here to raise an exception, or just return from the build.

File: readthedocs/projects/tasks.py [hunk 2]
```diff
@@ -90,6 +90,7 @@ def __init__(self, build_env=None, python_env=None, force=False, search=True, lo
         self.project = {}
         if project is not None:
             self.project = project
+        self.config = None
```
Comment (agjohnson) [inline readthedocs/projects/tasks.py]: We should replicate this in `run` as well

File: readthedocs/projects/tasks.py (hunk 1 above)
Comment (agjohnson) [inline readthedocs/projects/tasks.py]: I think returning is enough? There isn't anything to catch the `BuildEnvironmentError` here, and setup_env should set the build state correctly.

**Comment by agjohnson** [comment]
LGTM, pending CI pass

**Pull request merged**